crates/ruff_linter/src/rules/pylint/rules/unnecessary_assignment.rs

```diff
@@ -0,0 +1,199 @@
+use ruff_diagnostics::{Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{name::Name, Expr, ExprName, Stmt, StmtAssign, StmtIf};
+use ruff_python_codegen::Generator;
+use ruff_python_index::Indexer;
+use ruff_python_semantic::SemanticModel;
+use ruff_source_file::Locator;
+use ruff_text_size::Ranged;
+
+use crate::{checkers::ast::Checker, fix::edits::delete_stmt, settings::types::PythonVersion};
+
+/// ## What it does
+/// Check for cases where an variable assignment is directly followed by an if statement, these can be combined into a single statement using the `:=` operator.
+///
+/// ## Why is this bad?
+/// The code can written more concise, often improving readability.
+///
+/// ## Example
+///
+/// ```python
+/// test1 = "example"
+/// if test1:
+///     print(test1)
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// if test1 := "example":
+///     print(test1)
+/// ```
```

Comment:
```suggestion
/// ```

## References

- [Python Documentation: What’s New In Python 3.8](https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions)
- [PEP 572 – Assignment Expressions](https://peps.python.org/pep-0572/)
```

Reply:
Good suggestion! 😄